lua/starfall/libs_cl/midi.lua

```diff
@@ -0,0 +1,82 @@
+if not SF.Require("midi") then return function() end end
+
+--Midi Library
+-- @name midi
+-- @class library
+-- @libtbl midi_library
+SF.RegisterLibrary("midi")
+
+return function(instance)
+-- Only usable by the owner of the starfall chip
+if LocalPlayer() ~= instance.player then return end
+
+local midi_library = instance.Libraries.midi
+
+-- Event hook for midi
+-- @class hook
+-- @client
+-- @libtbl midi_library
+-- @param number time the exact systime which the event occured
+-- @param number command the command code of the event.  First 4 bits are the command code and last 4 are the channel
+-- @param varargs the different parameters of the midi event.  Can have 1 or more.
+SF.hookAdd("MIDI", "midi")
+
+-- Gets a table of all midi devices' ports.
+-- @return table table of all midi ports.
+function midi_library.getPorts()
+    return midi.GetPorts()
+end
+
+-- Opens the midi port to make it available to grab events from.  This must be called before the hook.
+-- @param number portNumber the midi port to open.
+-- @return string the name of the midi device opened at the given port.
+function midi_library.openPort(portNumber)
+    local portNumber = portNumber or 0
```

Comment:
Maybe want a checkluatype here (And all other functions that take params, idk if an error from the binary module would be cryptic)